### PR TITLE
Refactor survey endpoints to store parameters as JSON

### LINKED ISSUE #75: Refactor survey endpoints to use JSON blob
@dinah's post in slack:
```
/surveys
  POST - create survey object and store entire JSON blob
  PUT - update by overwriting the JSON blob
  SEND - turn JSON blob into models, then trigger send actions

Considerations:
1. Data is not easy to look through in its editing stage. No indexing.
2. Validations. If there is an error in the JSON blob and we don't validate
   until model creation then the user would not be notified until they try to
   send the survey.
3. Editing sent surveys. This setup makes sent surveys immutable objects. Once
   it is sent, the PUT endpoint is essentially useless because the JSON will not
   be serialized into objects again. (Might even make sense to disable PUT since
   we don't want discrepancies between the JSON blob and the associated survey
   objects.) In the human interpretation of what a survey is, does it makes
   sense for sent surveys to be immutable?
4. If we want to allow users to update surveys even after they've been sent, how
   can we support this? Solution: if we hit send again we re-serialize the JSON
   blob (which could have been updated since the last call to /surveys/:id/send)
   to create a new set of models that are distinct from the old models. Notice
   this would mean more than one survey object so there would need to be an
   interface (perhaps another lightweight object) introduced as the two objects
   cannot share the same id. It would be more like versioning of the same survey
   or, a closer model to our existing setup, create from draft.
```

### PR DESCRIPTION
**Pull request by dinahshi**
This PR only refactors existing create and send endpoints (i.e. does not include update). Adds a JSON column called parameters to surveys table. Moves creation of questions, response choices, question orders to survey model.

Fixes #75

### CODE REVIEW

File: app/models/survey.rb [hunk 1]
```diff
@@ -17,9 +18,66 @@ class Survey < ActiveRecord::Base
...
+  end
+
+  def generate_models
+    self.update!(
+      name: parameters['name'],
+      description: parameters['description'],
+    )
+
+    questions = generate_questions_and_response_choices(parameters['questions'])
+    self.update!(first_question: questions['1'])
```
Comment (chrisjluc) [inline app/models/survey.rb]: Shouldn't `'1'` be just `1`. Why are we accessing this hash as a string?

Edit: I just saw the tests they were changed from numbers to strings. What happened to the contract stating that these are numbers?

File: app/controllers/surveys_controller.rb [hunk 2]
```diff
@@ -13,6 +13,7 @@ def send_survey
       return render json: {error: :survey_not_found}, status: :not_found
     end
 
+    @survey.generate_models
```
Comment (chrisjluc) [inline app/controllers/surveys_controller.rb:17]: Can we have a check if the models are already generated don't generate them again? (And a test)

For the case where the user might want to send out more surveys for respondents they forgot. At this point the survey is immutable.

**Review by chrisjluc — CHANGES_REQUESTED** [review]
Looks pretty good! Just left a couple minor comments.

Wow github puts a giant X when I request changes, that's kind of intimidating lol.

**Comment by chrisjluc** [comment]
Also survey validation on POST / PUT can be a separate issue I suppose.

File: app/controllers/surveys_controller.rb (hunk 2 above)
Comment (dinahshi) [reply app/controllers/surveys_controller.rb:17]: What kind of check would suffice here? Any existing questions models associated with this survey means don't regenerate?

File: app/models/survey.rb (hunk 1 above)
Comment (dinahshi) [reply app/models/survey.rb]: My bad, wasn't sure if the 'number' defined in the interface was an integer or `'1'`. Will verify with @ayushbhagat and update.

File: app/controllers/surveys_controller.rb (hunk 2 above)
Comment (chrisjluc) [reply app/controllers/surveys_controller.rb:17]: Yeah, maybe if the question has a `first_question`. 

We can change it later if we need more strict requirements. This check may not be valid if we generate the models but they aren't valid, so we'd have to rollback the generation of those models. We can think about it later.

Comment (dinahshi) [reply app/controllers/surveys_controller.rb:17]: This is going to be much easier once we add the published flag so I'm going to add a TODO for now and come back to it.

**Review by chrisjluc — APPROVED** [review]

**Comment by chrisjluc** [comment]
LGTM!

**Pull request merged**